\documentclass[12pt]{phage3slides} %

\logo{\includegraphics[width=1cm]{gga-small.png}}
\title[Galaxy for Genome Annotation, Teaching, Databases]{Galaxy for Genome Annotation: Galaxy and GMOD for Annotation, Teaching, and Genomic Databases}
\author[ER, BG, ND, AB]{Eric Rasche, Bj\"orn Gr\"uning, Nathan Dunn, Anthony Bretaudeau}
\date{2017-06-29T13:50:00Z}

\begin{document}
\frame{\titlepage}

% GMOD projects have long provided powerful open-source tools to the
% bioinformatics community, but have historically been hard to configure
% and integrate. The Galaxy Genome Annotation (GGA) group provides a
% highly integrated set of Dockerized GMOD projects allowing for more
% widespread use of these tools in new contexts for system
% administrators wishing to deploy the suite. Our projects include
% maintenance of the Galaxy-Apollo bridge tools, Galaxy-Tripal and Chado
% tooling, and containerized versions of various GMOD projects which are
% configured to easily integrate with the rest of the suite.

% This talk will explore the use of this suite in the context of a real
% life use-case, an undergraduate phage annotation course. We will cover
% the GGA suite as well as various integrations, workflows, training
% materials, and tools that were built and made available in support of
% GGA.

% Very interesting subject. We encourage the presenters to focus on 3-4 key
% aspects of the project rather than try to cover every aspect.

\section[GGA]{Galaxy for Genome Annotoation}
\begin{frame}{Galaxy for Genome Annotation}
    Galaxy is great for:
    \begin{itemize}
        \item NGS Analysis
        \item Assembly
        \item ...
        \item Annotation Analysis (Tabular processing, etc)
        \item Proteomics, Metagenomics, *omics
    \end{itemize}
    But we are missing the Annotation step.\\\ \\
    We are missing the tooling, the
    trainings, and the community for genome annotation.
\end{frame}

\subsection{Who}
\begin{frame}{Who are we?}
    \begin{itemize}
        \item Bj\"orn Gr\"uning
        \item CPT Team (Eric Rasche, Eleni Mijalis, Cory Maughmer)
        \item Anthony Bretaudeau
        \item Peter van Heusen
        \item Nathan Dunn
        \item Eduardo de Paiva Alves
        \item Suzanna Lewis
        \item Torsten Seemann
        \item (\ldots you!)
    \end{itemize}
\end{frame}

\subsection{What}
\begin{frame}{What are we building?}
    \begin{itemize}
        \item Galaxy Images
        \item GMOD Containers
        \item Glue Code
        \item Training Materials
        \item Peripherals
        \item Community
    \end{itemize}
\end{frame}

\subsection{Why}
\begin{frame}{Why?}
    \begin{itemize}
        \item Fun Challenges
        \item Useful to real-life people, solve real problems
        \item Project longevity
    \end{itemize}
\end{frame}





\section{Annotation}
\begin{frame}{Annotation Infrastructure}
    \begin{itemize}
        \item Easy to deploy
        \item Shareable
        \item Configurable, pick-and-choose subcomponents
    \end{itemize}
\end{frame}

\subsection{Solutions}
\begin{frame}{Annotation Solutions}
    \begin{itemize}
        \item Docker Image: Galaxy + Annotation Tools {\color{gray}(Apollo Tools, Tripal Admin Tools, Circos, JBrowse, BLAST+, InterProScan, Glimmer, Augustus, FASTA manipulation tools, Spades, Mira, CD-Hit, ClustalW, AntiSmash, mummer, EMBOSS, BLAST, Diamond, Blast2GO)}
        \item Dockerized GMOD Deployment {\color{gray}(Galaxy, JBrowse, Apollo, Chado, Chado APIs, Tripal pre-configured to work together seamlessly)}
        \item Apollo, Chado python libraries {\color{gray}(+parsec like tools, ``Arrow'' and ``Chakin'')}
        \item Various Apollo support projects {\color{gray}(git-backup, experimental google docs integration)}
    \end{itemize}
\end{frame}

\subsection{Galaxy-Apollo}
\begin{frame}{Galaxy / Apollo Bridge}
    \begin{itemize}
        \item Grew out of use in an undergraduate course
        \item Initially quite simple, tool to add organism (JBrowse instance) to Apollo
        \item Expanded to include automation, tools for creating/updating/removing annotations.
        \item Jenkins job re-runs functional annotation workflows regularly
        \item Improved with student help, finding unexpected bugs
    \end{itemize}
\end{frame}









\section{Teaching}
\begin{frame}{Bacteriophage Annotation Course}
    \begin{itemize}
        \item Undergraduate Phage annotation course
        \item Genome sequence to publication
        \item Parallel track for environmental sample to isolated phage
        \item Novel genome, de novo annotation
        %\item Students run tools in Galaxy, annotate in Apollo, ``published'' data is stored in Chado
        %\item Converging implementations; updates to our phage course + GGA suite
    \end{itemize}
\end{frame}
%\item ``Google Docs'' of genome annotation
%\item Student+Faculty collaboration is easy
%\item Faculty+Faculty collaboration is easy

\subsection{Current}
\begin{frame}{Current Applications of GGA}
    \begin{itemize}
        \item Great example of GGA in real life
        \item CPT can leverage the GGA infrastructure, easily deploying Galaxy+Apollo+Chado
        \item GGA tools help the CPT build phage annotation workflows.
    \end{itemize}
\end{frame}

\subsection{Expansion}
\begin{frame}{Planned Expansions}
    % Too personal to CPT implementation?
    CPT's use case:
    \begin{itemize}
        \item Adding on Tripal + JBrowse-based public sequence display
        \item Building a sample tracking database, can leverage GGA Apollo / Chado, and Galaxy integrations
    \end{itemize}
    GGA:
    \begin{itemize}
        \item More tutorials / training resources
        \item More GMOD projects
    \end{itemize}
\end{frame}





\section{Databases}

\begin{frame}{Genomic Databases}
    \begin{itemize}
        \item Galaxy tools for Genomic Access
        \item Reproducibility for generally unreproducible external databases
        \item Democratization of resources%. (Easier access control through galaxy tools)
        %\item But how to democratize annotation?
        \item Galaxy Tools handle CRUD operations \& updating permissions
        %\item (But we do end up having to check user's apollo permissions at every step)
    \end{itemize}
\end{frame}

\subsection{Tooling}
\begin{frame}{Genomic Databases}
    \begin{itemize}
        \item Tools for querying annotation resources, answering specific questions. % TODO: integrate chado tools
        \item Tools for fetching remote data into Galaxy
        \item Galaxy is used for Analysis
        \item Tools for creating new annotations from analysis results
    \end{itemize}
\end{frame}





\section{Q\&A}
\begin{frame}{Q\&A}
    Thank you \\\ \\
    \begin{center}
        \begin{tabular}{ll}
            \color{gray} GGA GitHub & \href{https://galaxy-genome-annotation.github.io/}{galaxy-genome-annotation.github.io}\\
            \color{gray} GGA Gitter & \href{https://gitter.im/galaxy-genome-annotation/Lobby}{gitter.im/galaxy-genome-annotation/Lobby}\\
            \end{tabular}\\[1cm]
            %\logosTamuCPT
            \fundingNSFABIannotation
    \end{center}
\end{frame}

\end{document}
