\documentclass[12pt]{phage3slides} %

\logo{\includegraphics[width=1cm]{gga-small.png}}
\title[Galaxy for Genome Annotation, Teaching, Databases]{Galaxy Genome Annotation: Galaxy and GMOD for Annotation, Teaching, and Genomic Databases}
\author[ER, BG, ND, AB]{Eric Rasche, Bj\"orn Gr\"uning, Nathan Dunn, Anthony Bretaudeau}
\date{2017-06-29T13:50:00Z}

\begin{document}
\frame{\titlepage}

% GMOD projects have long provided powerful open-source tools to the
% bioinformatics community, but have historically been hard to configure
% and integrate. The Galaxy Genome Annotation (GGA) group provides a
% highly integrated set of Dockerized GMOD projects allowing for more
% widespread use of these tools in new contexts for system
% administrators wishing to deploy the suite. Our projects include
% maintenance of the Galaxy-Apollo bridge tools, Galaxy-Tripal and Chado
% tooling, and containerized versions of various GMOD projects which are
% configured to easily integrate with the rest of the suite.

% This talk will explore the use of this suite in the context of a real
% life use-case, an undergraduate phage annotation course. We will cover
% the GGA suite as well as various integrations, workflows, training
% materials, and tools that were built and made available in support of
% GGA.

% Very interesting subject. We encourage the presenters to focus on 3-4 key
% aspects of the project rather than try to cover every aspect.

\section[GGA]{Galaxy for Genome Annotoation}
\begin{frame}{Galaxy for Genome Annotation}
    Galaxy has room to expand into the Genome Annotation space.
    Needs community, we are that community.
    Needs tooling, we're building that.
    Needs integrations, we're building that.
    \begin{center}
        \includegraphics[width=0.5\textwidth]{gga-clean.png}
    \end{center}
\end{frame}

\subsection{Who}
\begin{frame}{Who are we?}
    \begin{itemize}
        \item Bj\"orn Gr\"uning
        \item Eric Rasche
        \item Anthony Bretaudeau
        \item Peter van Heusen
        \item Nathan Dunn
        \item Eduardo de Paiva Alves
        \item Suzanna Lewis
        \item Torsten Seemann
        \item (\ldots you!)
    \end{itemize}
\end{frame}

\subsection{What}
\begin{frame}{What are we building?}
    \begin{itemize}
        \item Docker Image: Galaxy + Annotation Tools {\color{gray}(Apollo Tools, Tripal Admin Tools, Circos, JBrowse, BLAST+, InterProScan, Glimmer, Augustus, FASTA manipulation tools, Spades, Mira, CD-Hit, ClustalW, AntiSmash, mummer, EMBOSS, BLAST, Diamond, Blast2GO)}
        \item Dockerized GMOD Deployment {\color{gray}(Most GMOD projects pre-configured to work together seamlessly)}
        \item Apollo Client library {\color{gray}(+parsec like tool, ``Arrow'')}
        \item Various Apollo support projects {\color{gray}(git-backup, experimental google docs integration)}
    \end{itemize}
\end{frame}

\subsection{Why}
\begin{frame}{Why?}
    \begin{itemize}
        \item Fun Challenges
        \item Useful to real-life people, solve real problems
    \end{itemize}
\end{frame}





\section{Annotation}
\begin{frame}{Annotation Infrastructure}
    \begin{itemize}
        \item Docker Image: Many annotation tools collected in a single place, pre-installed and ready to annotate
        \item Easy to deploy connections to annotation backends (e.g. Apollo)
        \item Peripheral tools to move data easily across annotation system boundaries %(Extract from Chado -> Load to Galaxy -> Analyse -> Push to Apollo)
    \end{itemize}
\end{frame}

\begin{frame}{Galaxy / Apollo Bridge}
    \begin{itemize}
        \item Initially simple tooling to move data in/out of Apollo
        \item Expanded to include annotation automation, tools for creating / removing ``human-created'' annotations.
        \item Jenkins scripts ran functional annotation workflows automatically, regularly.
        \item The Galaxy-Apollo bridge grew out of use in an undergraduate course
        \item We have improved upon them as students find new bugs and corner cases we did not expect
    \end{itemize}
\end{frame}









\section{Teaching}
\begin{frame}{Course}
    \begin{itemize}
        \item Course Background
        \item Current Course Applications
        \item Ongoing Expansions
    \end{itemize}
\end{frame}

\subsection{Background}
\begin{frame}{Course Background}
    \begin{itemize}
        \item Undergraduate Phage annotation course
        \item Genome sequence to publication
        \item (Parallel track for environmental sample to isolated phage)
        \item Gigantic workflows using many of the tools pre-installed in image
        \item Converging implementations; updates to our phage course + GGA suite
    \end{itemize}
\end{frame}

\subsection{Current}
\begin{frame}{Current Applications}
    \begin{itemize}
        \item Apollo for annotation is amazing, ``google docs'' of genome annotation
        \item Student+Faculty collaboration is easy
        \item Faculty+Faculty collaboration is easy

        \item Workflows (publicly available) for Phage Annotation
        \item More workflows available from training-material repo for 
    \end{itemize}
\end{frame}

\subsection{Expansion}
\begin{frame}{Applications}
    \begin{itemize}
        % Too personal to CPT implementation?
        \item Using automated Chado export in Apollo
        \item Adding on Tripal + JBrowse-based public sequence display
        \item Internal sample tracking database
        \item More GMOD projects to come?
    \end{itemize}
\end{frame}





\section{Database}

\begin{frame}{Genomic Databases}
    \begin{itemize}
        \item Perfect match
        \item Reproducibility + generally unreproducible external databases
        \item Democratization of resources. (Easier access control through galaxy tools)

        \item But how to democratize annotation?
        \item Galaxy Tools can create genomes, update permissions
        \item (But we do end up having to check user's apollo permissions at every step)
    \end{itemize}
\end{frame}

\subsection{Tooling}
\begin{frame}{Genomic Databases}
    \begin{itemize}
        \item Tools for querying annotation resources, answering specific questions. % TODO: integrate chado tools
        \item Tools for fetching remote data into Galaxy
        \item Galaxy is used for Analysis
        \item Tools for creating new annotations from analysis results
    \end{itemize}
\end{frame}





\section{Q\&A}
\begin{frame}{Q\&A}
    Thank you \\\ \\
    \begin{center}
        \begin{tabular}{rl}
            \color{gray} GGA GitHub & \href{https://galaxy-genome-annotation.github.io/}{galaxy-genome-annotation.github.io}\\
            \color{gray} GGA Gitter & \href{https://gitter.im/galaxy-genome-annotation/Lobby}{gitter.im/galaxy-genome-annotation/Lobby}\\
            \end{tabular}\\[1cm]
            %\logosTamuCPT
            \fundingNSFABIannotation
    \end{center}
\end{frame}

\end{document}
